common/autoware_overlay_rviz_plugin/autoware_minimap_overlay_rviz_plugin/src/minimap.cpp

```diff
@@ -0,0 +1,498 @@
+{
+  route_points_msg_ = msg;
+
+  // Convert the route points to local coordinates and set them in the RouteOverlay
+  std::vector<PathPoint> path_points;
+  for (const auto & pose : msg->points) {
+    PathPoint point;
+    point.x = pose.pose.position.x;
+    point.y = pose.pose.position.y;
+    path_points.push_back(point);
+  }
+
+  double origin_lat = property_origin_lat_->getFloat();
+  double origin_lon = property_origin_lon_->getFloat();
+  path_overlay_.setPathPoints(path_points, origin_lat, origin_lon);
+  path_overlay_.setVehiclePosition(latitude_, longitude_);
+
+  queueRender();
+}
+
+void VehicleMapDisplay::routeStateCallback(
+  const autoware_adapi_v1_msgs::msg::RouteState::SharedPtr msg)
+{
+  route_state_msg_ = msg;
+
+  queueRender();
+}
+
+// void VehicleMapDisplay::poseCallback(const nav_msgs::msg::Odometry::SharedPtr msg)
+void VehicleMapDisplay::poseCallback(
+  const autoware_adapi_v1_msgs::msg::VehicleKinematics::SharedPtr msg)
+{
+  pose_msg_ = msg;
+
+  latitude_ = msg->geographic_pose.position.latitude;
+  longitude_ = msg->geographic_pose.position.longitude;
+  target_latitude_ = latitude_;
+  target_longitude_ = longitude_;
+
+  property_longitude_->setFloat(target_longitude_);
+  property_latitude_->setFloat(target_latitude_);
+
+  goal_pose_.setVehiclePosition(target_latitude_, target_longitude_);
+  path_overlay_.setVehiclePosition(target_latitude_, target_longitude_);
+
+  updateMapPosition();  // Method to update map position based on new latitude and longitude
+}
+
+void VehicleMapDisplay::goalPoseCallback(const geometry_msgs::msg::PoseStamped::SharedPtr msg)
+{
+  goal_pose_msg_ = msg;
+
+  double origin_lat = property_origin_lat_->getFloat();
+  double origin_lon = property_origin_lon_->getFloat();
+  goal_pose_.setGoalPosition(msg->pose.position.x, msg->pose.position.y, origin_lat, origin_lon);
+
+  // Set the vehicle position in the goal pose (in case it was not set yet)
+  goal_pose_.setVehiclePosition(target_latitude_, target_longitude_);
+  path_overlay_.setVehiclePosition(target_latitude_, target_longitude_);
+
+  property_goal_lat->setFloat(goal_pose_.getGoalLatitude());
+  property_goal_lon->setFloat(goal_pose_.getGoalLongitude());
+  queueRender();
+}
+
+std::pair<double, double> VehicleMapDisplay::localXYZToLatLon(double x, double y)
+{
+  int zone;
+  bool north_p;
+  double origin_lat = property_origin_lat_->getFloat();
+  double origin_lon = property_origin_lon_->getFloat();
+  double origin_x, origin_y, gamma, k;
+
+  // Convert origin to UTM coordinates
+  GeographicLib::UTMUPS::Forward(
+    origin_lat, origin_lon, zone, north_p, origin_x, origin_y, gamma, k);
+
+  // Calculate global UTM coordinates by adding local offsets
+  double global_x = origin_x + x;
+  double global_y = origin_y + y;
+
+  // Convert back to geographic coordinates
+  double lat, lon;
+  GeographicLib::UTMUPS::Reverse(zone, north_p, global_x, global_y, lat, lon);
+
+  return {lat, lon};
```

Comment:
This projection depends on how the map is loaded in Autoware.
You should be able to retrieve the information from /map/map_projector_info topic and switch the projection to support different projection methods.